src/dash-table/utils/actions.js

```diff
@@ -53,23 +61,36 @@ export const clearSelection = {
     selected_cells: []
 };
 
-export function editColumnName(column, columns, headerRowIndex) {
+export function changeColumnHeader(column, columns, headerRowIndex, mergeDuplicateHeaders, newColumnName) {
+    let cloneColumn = Object.assign({}, column);
+    if ((typeof column.name === 'string') || (column.name instanceof String)) {
+            const columnHeaders = columns.map(col => col.name);
+            const maxLength = findMaxLength(columnHeaders);
+            const newColumnNames = Array(maxLength).fill(column.name);
+            cloneColumn = Object.assign({}, column, {name: newColumnNames});
+    }
+    const transformedColumns = columns.map(col => {
+        if (((typeof column.name === 'string') || (column.name instanceof String)) && col.id === column.id) {
```

Comment:
`instanceof String` is only useful if we use the `new String(...)`, which we do not (guaranteed through tslint no-construct rule), `typeof` should be enough.